Add additional attrs to shortcut

cheunghy opened the pull request:
This commit add additional attributes to shortcut.

Useful when output complicated tag with simplified shortcut
``` ruby
shortcut: {'^' => {tag: 'script', attr: 'data-binding', attrs: { type: "application/json" }}}
```
Then user input
``` slim
^items
  == "[{'title':'item0'},{'title':'item1'}]
```
will output
``` html
<script data-binding="items" type="application/json">[{'title':'item0'},{'title':'item1'}]</script>
```

Unit test added, too.

minad commented:
I would prefer if you just make :attr a hash. So that it can be either a string or a hash. What do you think?

cheunghy commented:
attr means that one:
```.abc```
For this, attr means the key for the shortcut class,
so ```.abc``` becomes class="abc"
If we make attr a hash, then maybe {shortcut: "class"}, but how about user has additional shortcut attribute? If make it name like {__shortcut: "class"}, it's not good looking.

minad commented:
Ah sure, you are right. Then we can keep it like this. But I would still prefer another name which doesn't only differ from attr by one letter. Maybe something like :additional_attrs? Could you also update the readme and the changelog please?

cheunghy commented:
Thanks.
I rename the key to additional_attrs, updated changelog and readme.
And also set the version to 3.0.7 (was 3.0.6).

minad commented:
Ok, but no version update please.

> Thanks.
> I rename the key to additional_attrs, updated changelog and readme.
> And also set the version to 3.0.7 (was 3.0.6).

cheunghy commented:
I removed version update. 
I updated version just because you asked me to update change log, and change log doesn't have unversioned ones.

I just put 2015-09-04 as change log title.

dmke commented on test/core/test_html_structure.rb:
```diff
@@ -7,7 +7,7 @@ def test_simple_render
 html
   head
     title Simple Test Title
-  body 
```
Please leave this trailing space, it's actually a test case for #82.

It'd be great if you also could add a line like

```ruby
assert_match source, /body\s+$/, "missing test case for issue #82"
```

cheunghy commented:
@dmke I've added back the whitespace.

@minad Through another commit, I add test case for issue #415

minad commented:
@cheunghy thx!
